pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/InternalConfiguration.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.admin;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import org.apache.bookkeeper.conf.ClientConfiguration;
+import org.apache.pulsar.common.conf.InternalConfigurationData;
+
+@Path("/internal-configuration")
+@Api(value = "/internal-configuration", description = "InternalConfiguration admin apis", tags = "internal-configuration")
+@Produces(MediaType.APPLICATION_JSON)
+public class InternalConfiguration extends AdminResource {
+
+    @GET
+    @ApiOperation(value = "Get the internal configuration data", response = InternalConfigurationData.class)
+    @ApiResponses(value = {
+        @ApiResponse(code = 403, message = "Don't have admin permission")
```

Comment:
in which scenario it returns 403?

Reply:
isn't accessing admin rest endpoint requires authentication? 403 is returned when client is not authenticated to access admin rest endpoint, no?

please correct me if I am wrong.

Reply:
yes, that's correct. I meant `code = 403, message = "Don't have admin permission"` if we think zk-endpoint information should not expose to user and should be accessed by only admin then we can  add validation `AdminResource::validateSuperUserAccess()`